index.js

```diff
@@ -1,109 +1,145 @@
 #!/usr/bin/env node
 
-import fs from 'fs';
-import path from 'path';
+import 'babel-polyfill';
+import program from 'commander';
+import { stat, readFile, writeFile, unlink } from 'fs-extra';
 import { compileToString } from 'node-elm-compiler';
+import path from 'path';
 import sh from 'shelljs';
 
-const filename = (process.argv[2] || '').trim();
+(async () => {
+  program
+    // package is one level up because the compiled js file is in lib/
+    // eslint-disable-next-line import/no-unresolved
+    .version(require(path.resolve(__dirname, '../package.json')).version)
+    .usage('[options] <file>')
+    .option(
+      '--output-var [name]',
+      'name of the elm variable to output (should be of type String)',
+      'output'
+    )
+    .option(
+      '--project-dir [path]',
+      'specific directory in which to search for elm-package.json or to create it (if different from file location)'
+    )
+    .parse(process.argv);
+
+  // run-elm expect exactly one argument, so exit otherwise
+  if (program.args.length !== 1) {
+    console.log(program.help());
+    process.exit(1);
+  }
 
-if (!filename || filename === '-h' || filename === '--help') {
-  console.log('Usage: run-elm [FILENAME]');
-  process.exit(1);
-}
+  // read args and options
+  const rawUserModuleFileName = program.args[0];
+  const outputVarName = program.outputVar;
+  const rawProjectDir = program.projectDir;
+
+  // extract key paths
+  const timestamp = +new Date();
+  const mainModule = `RunElmMain${timestamp}`;
```

Comment:
With a timestamp in the end of the main module name, we can avoid naming collisions and race conditions.